File: challenge-372/abigail/go/ch-1.go
package main

import (
    "fmt"
    "bufio"
    "os"
    "strings"
)

func main () {
    reader := bufio . NewReader (os . Stdin)
    for {
        text, err := reader . ReadString ('\n')
        if (err != nil) {
            break
        }
        
        text     = strings . ReplaceAll (text, "\"", "")  // Remove quotes
        text     = strings . ReplaceAll (text, "\n", "")  // Remove newline
        spaces  := strings . Count      (text, " ")       // Count spaces
        words   := strings . Fields     (text)            // Split into words
        gaps    := len (words) - 1                        // Nr of gaps

        per_gap := 0                                      // Calculate amount
        if gaps > 0 {                                     //   of spaces needed
            per_gap = spaces / gaps
        }
        end     := spaces - gaps * per_gap

        fmt . Printf ("\"")                               // Print quote
        for index, value := range (words) {
            fmt . Printf ("%s", value)                    // Print word
            max := end
            if index < len (words) - 1 {
                max = per_gap
            }
            for i := 0; i < max; i ++ {                   // Print gap,
                fmt . Printf (" ")                        //   space by space
            }
        }
        fmt . Printf ("\"\n")                             // Print quote
    }
}

File: challenge-372/abigail/node/ch-1.js
#!/opt/homebrew/bin/node
 
  require ('readline')
. createInterface ({input: process . stdin})
. on              ('line', line => {
    //
    // Count spaces
    //
    let spaces  = [... line . matchAll (/ /g)] . length

    //
    // Remove quotes, and split into words. When splitting,
    // Javascript retains empty leading/trailing fields, so 
    // we need to filter them.
    //
    let words   = line  . replaceAll (/"/g, "")
                        . split (/\s+/)
                        . filter (x => x . length > 0)
    let gaps    = words . length - 1

    //
    // Calculate spaces needed
    //
    let per_gap = gaps ? Math . floor (spaces / gaps) : 0
    let end     = spaces - gaps * per_gap

    //
    // Print results
    //
    console . log ('"' + words . join (" " . repeat (per_gap))
                       +               " " . repeat (end) + '"')
  })
